### DISCUSSION TITLE
ProjectTracker blazor server - dal is null

### DISCUSSION THREAD

**Discussion by geco3** [MarimerLLC/csla#2599] [Questions] [answered]
Hello,

Long time CSLA user, first time Blazor user. Downloaded CSLA 5.5.1 and opened the ProjectTracker solution and configure ProjectTracker.Ui.Blazor.Server +AppServer as startup. 

      //// use in-proc data portal
      app.UseCsla();

The downloaded code uses the in-proc data portal by default with remote data portal server commented out so I guess there is not requirement to startup the AppServer.

On application start get exception-

System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=ProjectTracker.BusinessLibrary
  StackTrace:
   at ProjectTracker.Library.RoleList.Fetch(IRoleDal dal) in E:\Projects\CSLA\csla-5.5.1\Samples\ProjectTracker\ProjectTracker.BusinessLibrary.Netstandard\RoleList.cs:line 72

If I change the code to use the data portal it works with the already logged issue on authentication - Unable to cast object of type 'Csla.Security.UnauthenticatedPrincipal' to type 'System.Security.Claims.ClaimsPrincipal'.

What am I missing to get the in-proc data portal working?

Thanks,

Geco.

**Comment by rockfordlhotka** [comment, marked as answer]
Hi,

I suggest you start with the BlazorExample project first, because it is simpler than ProjectTracker, and it sounds like you are trying to get into Blazor, so that might be a better start point.

Sadly, CSLA 5 has issues with server-side Blazor. Those issues are being addressed in CSLA 6, because the fix is very deep and requires breaking changes. I discuss them in some depth in this thread: https://github.com/MarimerLLC/csla/discussions/2595

**Reply by geco3** [reply]
Thanks Rocky.
